### ISSUE TITLE
read_packager_file doesn't allow setting module_name

### ISSUE THREAD

**Issue by carlwgeorge** [fedora-modularity/libmodulemd#535]
The CentOS team routinely needs to read modulemd files with no name ([example](https://git.centos.org/modules/virt/blob/98fad7bdd0e4c84796ab04535d08dcf516333a34/f/SOURCES/modulemd.src.txt)).  We currently do that using `Modulemd.ModuleStream.read_file` in Python, which allows us to pass in an optional module name argument.

```python
source_modulemd = Modulemd.ModuleStream.read_file('SOURCES/modulemd.src.txt', True, module_name)
```

Since the 2.11 update on my Fedora workstation, I now get a deprecation warning.  The [documentation](https://fedora-modularity.github.io/libmodulemd/latest/ModulemdModuleStream.html#modulemd-module-stream-read-file) says that the recommended replacement is `Modulemd.read_packager_file`.  However, that method does not allow setting the module name during invocation.  Can `Modulemd.read_packager_file` add the ability to pass in an optional module name?

Edit: I spoke with @sgallagher about this and he asked me to provide more context.  The CentOS team has to rebuild RHEL modules based on the modulemd.src.txt files that are exported to https://git.centos.org.  Those files must be passed through a translation tool that converts the internal RHEL git branch names to public git commit references in https://git.centos.org.  That translation tool attempts to add the missing name field, as well as naming the new file using the module name for the expected input to our MBS instance.  That is why we require the ability to manually specify the module name when reading the modulemd.src.txt file.  It exists in the deprecated method, and we would like it to exist in the new method so it is a proper replacement for our needs.

**Comment by mmathesius** [comment]
Something like that should be possible--at least from the Python perspective. However, we need to be careful changing the existing API in the C code.

@mcurlej @sgallagher What are your thoughts on this RFE?

P.S. With the year-end holidays upon us, don't expect an movement on this before the beginning of the new year.

**Comment by mmathesius** [comment]
I had a few spare moments so I worked on this a bit. As noted in the associated draft PR, I have provided the C implementation to go along with this request, but I don't know how to integrate it with the Python API.

**Comment by mcurlej** [comment]
@mmathesius I am ok with this, just please make it so it is the same API as in `Modulemd.ModuleStream.read_file` so it is not mandatory but only optional.

**Comment by mmathesius** [comment]
I figured out how to integrate the enhanced functionality into the Python API--with the extra arguments being optional. See the additional commit in my PR.

But I have a follow-up question: Should the enhanced versions of the functions also include a `strict` parameter like `Modulemd.ModuleStream.read_file()` and `Modulemd.ModuleStream.read_string()` have? I'm not sure it's even meaningful to use with `read_packager_file()`.

**Comment by mmathesius** [comment]
Fixed by #536.
